### PR TITLE
General updates

### PR DESCRIPTION
**Pull request by cz4rs**
Follow-up to https://github.com/ECP-copa/CabanaMD/pull/113#issuecomment-2321392846.

General changes extracted from load balancing work in #113 and https://github.com/DARMA-tasking/CabanaMD/pull/3.

### CODE REVIEW

File: src/integrator_nve_impl.h [hunk 1]
```diff
@@ -64,10 +64,8 @@ void Integrator<t_System>::initial_integrate( t_System *system )
     f = system->f;
     type = system->type;
 
-    static int step = 1;
```
Comment (cz4rs) [inline src/integrator_nve_impl.h:67]: `clang-18` gives me `variable set but not used` warning here.

**Comment by cz4rs** [comment]
@streeve Can you please have a look? I will post the second (load balancing) part once this goes through.

File: src/integrator_nve_impl.h (hunk 1 above)
Comment (streeve) [reply src/integrator_nve_impl.h:67]: I believe it was intended to help debugging in kernel - fine with me to remove

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -98,7 +98,7 @@ jobs:
         uses: actions/checkout@v3
         with:
           repository: ECP-CoPA/Cabana
-          ref: 0.6.1
+          ref: master
```
Comment (streeve) [inline .github/workflows/CI.yml]: We try to keep the CI on a tagged version and Nightly on master/develop, but if we need anything newer we can pick a hash

Comment (cz4rs) [reply .github/workflows/CI.yml]: Makes sense. I have rebased that change out, will switch to hash later if necessary.

**Review by streeve — APPROVED** [review]

**Pull request merged**